extensions/interactions/ItemSelectionInput/directives/oppia-interactive-item-selection-input.component.ts

```diff
@@ -13,124 +13,140 @@
+      'ItemSelectionInput',
+      {
+        choicesWithValue: this.choicesWithValue,
+        maxAllowableSelectionCountWithValue:
+          this.maxAllowableSelectionCountWithValue,
+        minAllowableSelectionCountWithValue:
+          this.minAllowableSelectionCountWithValue
+      }
+    ) as ItemSelectionInputCustomizationArgs;
+
+    this.choicesValue = choices.value;
+    this.choices = this.choicesValue.map(choice => choice.html);
+    this.maxAllowableSelectionCount = maxAllowableSelectionCount.value;
+    this.minAllowableSelectionCount = minAllowableSelectionCount.value;
+
+    // The following is an associative array where the key is a choice
+    // (html) and the value is a boolean value indicating whether the
+    // choice was selected by the user (default is false).
+    this.userSelections = {};
+
+    for (var i = 0; i < this.choices.length; i++) {
+      this.userSelections[this.choices[i]] = false;
     }
-  ]
-});
+
+    this.displayCheckboxes = this.maxAllowableSelectionCount > 1;
+
+    // The following indicates that the number of answers is more than
+    // maxAllowableSelectionCount.
+    this.preventAdditionalSelections = false;
+
+    // The following indicates that the number of answers is less than
+    // minAllowableSelectionCount.
+    this.notEnoughSelections = this.minAllowableSelectionCount > 0;
+    this.currentInteractionService.registerCurrentInteraction(
+      this.submitAnswer.bind(this), this.validityCheckFn.bind(this));
+  }
+
+  onToggleCheckbox(): void {
+    this.newQuestion = false;
+    this.selectionCount = Object.keys(this.userSelections).filter(
+      (obj) => this.userSelections[obj]).length;
+    this.preventAdditionalSelections = (
+      this.selectionCount >= this.maxAllowableSelectionCount);
+    this.notEnoughSelections = (
+      this.selectionCount < this.minAllowableSelectionCount);
+  }
+
+  submitMultipleChoiceAnswer(event: MouseEvent, index: number): void {
+    event.preventDefault();
+    // Deselect previously selected option.
+    var selectedElement = (
+      document.querySelector(
+        'button.multiple-choice-option.selected'));
+    if (selectedElement) {
+      selectedElement.classList.remove('selected');
+    }
+    // Selected current option.
+    (event.currentTarget as HTMLDivElement).classList.add('selected');
+    this.userSelections = {};
+    this.userSelections[this.choices[index]] = true;
+    this.notEnoughSelections = false;
+    if (!this.browserCheckerService.isMobileDevice()) {
+      this.submitAnswer();
+    }
+  }
+
+  submitAnswer(): void {
+    const htmlAnswers = Object.keys(this.userSelections).filter(
+      (obj) => this.userSelections[obj]);
+    const answers = htmlAnswers.map(
+      html => this.choicesValue[this.choices.indexOf(html)].contentId);
+
+    this.currentInteractionService.onSubmit(
+      answers as unknown as string,
+      // eslint-disable-next-line max-len
```

Comment:
We always think of disabling linting rules as the last resort.

Reply:
Remove